yabt/builders/cpp.py

```diff
@@ -253,9 +253,13 @@ def compile_cc(build_context, compiler_config, buildenv, sources,
         obj_rel_path = '{}.o'.format(splitext(src)[0])
         obj_file = join(buildenv_workspace, obj_rel_path)
         include_paths = [buildenv_workspace] + compiler_config.include_path
+
+        # ${buildenv_workspace} is replaced by the actual value at
+        # compilation time
         compile_cmd = (
             [compiler_config.compiler, '-o', obj_file, '-c'] +
-            compiler_config.compile_flags +
+            [flag.replace('${buildenv_workspace}',buildenv_workspace) for
+                            flag in compiler_config.compile_flags] +
```

Comment:
please add a short test for this - this is important also because tests document the functionality

Reply:
Done, see cpp_test.py